tests/transport_test.go

```diff
@@ -60,12 +60,21 @@ var _ = Describe("Transport Tests", func() {
 			sec *v1.Secret
 		)
 
+		var stringData map[string]string
 		if credentialed {
 			By(fmt.Sprintf("Creating secret for endpoint %s", ep))
-			stringData := map[string]string{
-				common.KEY_ACCESS: utils.AccessKeyValue,
-				common.KEY_SECRET: utils.SecretKeyValue,
+			if accessKey != "" && secretKey != "" {
+				stringData = map[string]string{
+					common.KEY_ACCESS: accessKey,
+					common.KEY_SECRET: secretKey,
+				}
+			} else {
+				stringData = map[string]string{
+					common.KEY_ACCESS: utils.AccessKeyValue,
+					common.KEY_SECRET: utils.SecretKeyValue,
+				}
 			}
+
```

Comment:
slightly cleaner, imo, to refactor as:
```
		var stringData map[string]string
		if credentialed {
			By(fmt.Sprintf("Creating secret for endpoint %s", ep))
                        if accessKey == "" {
                                      accessKey =  utils.AccessKeyValue
                        }
                        if secretKey == "" {
                                       secretKey == utils.SecretKeyValu
                        }
		       stringData[common.KEY_ACCESS] = accessKey
                       stringData[common.KEY_SECRET] = secretKey
...
```

Reply:
It may that credential is indicated only by `accessKey != ""` and the secret value can be ignored for this test?

Reply:
meaning, no secret? just accessKey? Is it po ssible? if so, I can change this

Reply:
okay

Reply:
command := `expSize=$(curl -s http://cdi-file-host.kube-system | grep -E '"name":"tinyCore.iso"' | grep -Eo '"size":[0-9]+' | tr -d '":[:alpha:]'); haveSize=$(wc -c < /pvc/disk.img); (( $expSize == $haveSize )); echo $?`

exitCode := f.ExecShellInPod(pod.Name, ns, command)
// A 0 exitCode should indicate that $expSize == $haveSize
Expect(strconv.Atoi(exitCode)).To(BeZero())

returns error:

Transport Tests
/go/src/kubevirt.io/containerized-data-importer/tests/transport_test.go:20
  Transport Test Table
  /go/src/kubevirt.io/containerized-data-importer/vendor/github.com/onsi/ginkgo/extensions/table/table.go:92
    should not connect to http endpoint with invalid credentials [It]
    /go/src/kubevirt.io/containerized-data-importer/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:46

    Expected
        <int>: 1
    to be zero-valued

    /go/src/kubevirt.io/containerized-data-importer/tests/transport_test.go:96